client/network/src/service.rs

```diff
@@ -2025,12 +2025,16 @@ where
 
 						if this.boot_node_ids.contains(&peer_id) {
 							if let DialError::WrongPeerId { obtained, endpoint } = &error {
-								error!(
-									"💔 The bootnode you want to connect provided a different peer ID than the one you expect: `{}` with `{}`:`{:?}`.",
-									peer_id,
-									obtained,
-									endpoint,
-								);
+								if let ConnectedPoint::Dialer { address, role_override: _ } =
```

Comment:
How about ConnectedPoint::Listener?

Reply:
We are handling here `SwarmEvent::OutgoingConnectionError` (on line 2018), so we are the ones who dialed the node. `Listener` is not possible in this branch.

Reply:
Well I agree with you. Can you file an issue in rust-libp2p repo?

Reply:
Sure. Do I understand correctly that `ConnectedPoint` in `DialError` should be replaced just with `Dialer` in rust-libp2p?

Reply:
Created the issue: https://github.com/libp2p/rust-libp2p/issues/2767. Could you approve the PR so that we have pretty error output before the issue is resolved in libp2p?